### PR TITLE
Rename maven coordinates

### LINKED ISSUE #241: Rename maven coordinates
As discussed we will be moving to `org.spekframework.spek2` as our group id. 

Below are the new coordinates:

- `org.jetbrains.spek:spek-api` -> `org.spekframework.spek2:spek-dsl` 
- `org.jetbrains.spek:spek-junit-platform-engine` -> `org.spekframework.spek2:spek-runner-junit5`
- `org.jetbrains.spek:spek-subject-extension` -> `org.spekframework.spek2:spek-extension-subject`
- `org.jetbrains.spek:spek-data-driven-extension` -> `org.spekframework.spek2:spek-extension-data-driven`

Here are some additional artifacts for `2.0`:

- `org.spekframework.spek2:spek-runtime-common`
- `org.spekframework.spek2:spek-runtime-jvm`

Do note that this not include renaming the packages.

### PR DESCRIPTION
**Pull request by raniejade**
Resolves #241.

### CODE REVIEW

File: distribution/build.gradle [hunk 1]
```diff
@@ -73,9 +70,9 @@ publishing {
             }
         }
 
-        subjectExtension(MavenPublication) {
-            groupId = 'org.jetbrains.spek'
-            artifactId = 'spek-subject-extension'
+        extensionSubject(MavenPublication) {
+            groupId = 'org.spekframework.spek2'
```
Comment (artem-zinnatullin) [inline distribution/build.gradle]: Let's extract groupid to common variable

Comment (raniejade) [inline distribution/build.gradle]: :+1:

Comment (raniejade) [inline distribution/build.gradle]: Latest change extracted `groupId` and `version`, it took advantage of the `maven-publish`'s default values.

Comment (artem-zinnatullin) [inline distribution/build.gradle]: Not sure I follow…

Comment (raniejade) [inline distribution/build.gradle]: Wrong change, sorry.

Comment (raniejade) [inline distribution/build.gradle]: should be fixed now.

**Comment by raniejade** [comment]
Merged manually to `2.x`.

**Pull request closed without merging**